dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -11,15 +11,18 @@ class Api::V1::MlModelsController < Api::V1::JsonApiController
   def save
     model_id = generate_id
     metadata = params["ml_model"].except(:trainedModel, :featureNumberKey)
-    UserMlModel.create!(
+    @user_ml_model = UserMlModel.create!(
       user_id: current_user.id,
       model_id: model_id,
       name: params["ml_model"]["name"],
       metadata: metadata.to_json
     )
-    upload_to_s3(model_id, params["ml_model"].to_json)
-
-    render json: {id: model_id}
+    if @user_ml_model.persisted?
+      upload_to_s3(model_id, params["ml_model"].to_json)
+      render json: {id: model_id, message: "The model successfully saved!"}
```

Comment:
I think we should have a error/success ID here so that the server is language-agnostic, and then the client can show the appropriate user-facing string for that ID.